app/Controllers/Timetable.hs

```diff
@@ -327,11 +333,14 @@ getDatesByDay session dataByDay
     | session == "F" = do
         fallStart <- fallStartDate
         fallEnd <- fallEndDate
-        formatDates $ getDates fallStart fallEnd (weekDay $ head dataByDay)
+        formatDates $ getDates fallStart fallEnd dayOfWeek
     | otherwise = do
         winterStart <- winterStartDate
         winterEnd <- winterEndDate
-        formatDates $ getDates winterStart winterEnd (weekDay $ head dataByDay)
+        formatDates $ getDates winterStart winterEnd dayOfWeek
+    where dayOfWeek = weekDay (case dataByDay of
```

Comment:
You can do top-level pattern matching here as well: `getDatesByDay session [] = ...`, etc.